### PR TITLE
Support for Message Forward

### PR DESCRIPTION
**Pull request by Doc94**
**Description:** this PR implements https://github.com/Discord4J/discord-json/pull/187 for allow the support for forward messages.
this also include a minor breaking change based in how the MessageCreateSpec just expect a Snowflake for the message reference what currently is inconsistent with the discord API for allow answers/forwards

### CODE REVIEW

File: core/src/main/java/discord4j/core/object/entity/Message.java [hunk 1]
```diff
@@ -702,6 +714,11 @@ public MessageEditMono edit() {
         return MessageEditMono.of(this);
     }
 
+    public Mono<Message> forward(TopLevelGuildMessageChannel channel) {
```
Comment (Doc94) [inline core/src/main/java/discord4j/core/object/entity/Message.java:717]: a little unsure about this method.. mostly if is valid just allow pass the channel... or add a helper method in channel like createForwardMessage with a Message like argument for autofill the spec

Comment (Azn9) [reply core/src/main/java/discord4j/core/object/entity/Message.java:717]: You can forward messages to DMs and private groups so just use a Snowflake as argument?

File: core/src/main/java/discord4j/core/spec/MessageCreateSpecGenerator.java [hunk 2]
```diff
@@ -62,7 +63,14 @@ default List<MessageCreateFields.FileSpoiler> fileSpoilers() {
...
     Possible<AllowedMentions> allowedMentions();
 
-    Possible<Snowflake> messageReference();
+    /**
+     * @deprecated this is just a helper for the reply format in this spec you need {@link MessageReferenceData}
+     */
+    @Deprecated
+    Possible<Snowflake> messageReferenceId();
+
+    @Experimental
```
Comment (Azn9) [inline core/src/main/java/discord4j/core/spec/MessageCreateSpecGenerator.java:72]: This shouldn't be experimental if the other method is deprecated

**Review by Azn9 — CHANGES_REQUESTED** [review]

File: core/src/main/java/discord4j/core/object/entity/Message.java (hunk 1 above)
Comment (Doc94) [reply core/src/main/java/discord4j/core/object/entity/Message.java]: Yeah.. i modify this method for allow just pass the MessageChannel with the createMessage method and also include a basic forward for just pass channel Id with the same behaviour for try to parse to the MessageChannel with the method to forward.

File: core/src/main/java/discord4j/core/spec/MessageCreateSpecGenerator.java (hunk 2 above)
Comment (Doc94) [reply core/src/main/java/discord4j/core/spec/MessageCreateSpecGenerator.java]: Removed

**Comment by Doc94** [comment]
> I think that if you tested the methods and they works as expected, the Experimental annotation is not needed

i wanna keep the experimental annotation for the getMessageSnapshots mostly because i just go directly to the Partial message data and if discord just change that then need rework this method again unless i just make a MessageSnapshot class with the just getPartialMessage but looks like a so "empty" subclass

**Comment by Azn9** [comment]
> i wanna keep the experimental annotation for the getMessageSnapshots mostly because i just go directly to the Partial message data and if discord just change that then need rework this method again unless i just make a MessageSnapshot class with the just getPartialMessage but looks like a so "empty" subclass

I think an "empty" class is not a huge deal if this can prevent reworks if discord change things

**Comment by Doc94** [comment]
> > i wanna keep the experimental annotation for the getMessageSnapshots mostly because i just go directly to the Partial message data and if discord just change that then need rework this method again unless i just make a MessageSnapshot class with the just getPartialMessage but looks like a so "empty" subclass
> 
> I think an "empty" class is not a huge deal if this can prevent reworks if discord change things

yeah i make that changes now if can review :3

**Review by Azn9 — CHANGES_REQUESTED** [review]
Just missing the copyright header on new files but else looks good for me

**Comment by Doc94** [comment]
> Just missing the copyright header on new files but else looks good for me

ready

**Pull request merged**